sympy/core/numbers.py

```diff
@@ -784,11 +788,14 @@ def __new__(cls, num, prec=None):
         elif num is S.NegativeInfinity:
             num = '-inf'
         elif isinstance(num, mpmath.mpf):
-            if prec == None:
-                prec = num.context.dps
+            if prec is None:
+                if dps is None:
+                    prec = num.context.prec
+            else:
+                prec = num.context.prec
```

Comment:
Ah, yes. I was just beginning to write some tests and I spotted this discrepancy.Thanks for pointing it out.